.github/workflows/Ci.yml

```diff
@@ -0,0 +1,109 @@
+name: "CI: Run tests"
+
+# GitHub Actions triggers based on the PR's and each commits on develop and master
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches: [develop, master,feature/add_gha]
+
+jobs:
+  build:
+    defaults:
+      run:
+        shell: bash
+        working-directory: "/opt/carma/"
+
+    # Specify the runner environment
+    runs-on: ubuntu-latest
+
+    # Set up a Docker container for the job
+    container:
+      image: usdotfhwastoldev/autoware.ai:develop
+      env:
+        INIT_ENV: "/home/carma/.base-image/init-env.sh"
+        ROS_2_ENV: "/opt/ros/foxy/setup.bash"
+        TERM: xterm
+      options: "--user root"
+
+    steps:
+    - name: Checkout ${{ github.event.repository.name }}
+      # Check out the repository code
+      uses: actions/checkout@v3.3.0
+      with:
+        fetch-depth: 0
+        path: src/${{ github.event.repository.name }}
+
+    - name: Move source code
+      # Move the source code to the desired location
+      run: mv $GITHUB_WORKSPACE/src /opt/carma/
+
+    - name: Checkout dependencies
+      # Checkout project dependencies
+      run: |
+        source "$INIT_ENV"
+        ./src/${{ github.event.repository.name }}/docker/checkout.bash -r /opt/carma/
+
+    - name: Install external dependencies
+      # Install the multiple object tracking deps
+      run: sudo bash /opt/carma//src/multiple_object_tracking/scripts/install_dependencies.sh
+
+    - name: Build ROS1
+      # Build ROS1 packages
+      # The skipped ROS 1 packages here should match the ROS 1 packages that are skipped during the build in docker/install.sh
+      run: |
+        source "$INIT_ENV"
+        PACKAGES=$(find . -maxdepth 2 -type f -name package.xml | sed 's/\.\///' | cut -d/ -f1)
+        sed -i 's/colcon build /colcon build --packages-skip novatel_oem7_msgs tracetools tracetools_test /' /home/carma/.ci-image/engineering_tools/code_coverage/make_with_coverage.bash
+        make_with_coverage.bash -m -e /opt/carma/ -o ./coverage_reports/gcov
+
+    - name: Run ROS1 C++ Tests
+      # Run ROS1 C++ tests
+      # The skipped ROS 1 packages here should match the ROS 1 packages that are skipped during the build in docker/install.sh"
+      run: |
+        source "$INIT_ENV"
+        sed -i 's/colcon test /colcon test --packages-skip novatel_oem7_msgs tracetools tracetools_test carma_launch_testing carma_message_utilities ros2bag ros2trace tracetools_launch tracetools_read tracetools_trace multiple_object_tracking platoon_control platoon_control_ihp rosbag_mock_drivers /' /home/carma/.ci-image/engineering_tools/code_coverage/make_with_coverage.bash

```

Comment:
Let's remove the listed packages here for all packages that are not skipped in `docker/install.sh`. This will cause the workflow to fail, but we will address that in separate work (https://usdot-carma.atlassian.net/browse/CAR-6008).

Reply:
EDIT: I might have been partially wrong about this. The "CI: Run Tests" workflow now shows failing unit tests for the packages we skipped during the `Build ROS1` step. Maybe we actually need to add the `sed` command back in here to skip the unit tests of those packages (just `novatel_oem7_msgs`, `tracetools`, and `tracetools_test`)?